### DISCUSSION TITLE
How to output depth information to SITL?

### DISCUSSION THREAD

**Discussion by BrellenZhou** [microsoft/AirSim#3643] [Support Q&A] [answered]
Hi, 

I want to output airsim's depth information to Ardupilot in SITL to achieve obstacle avoidance target. Ardupilot/PX4 has already have the computer-vision based obstacle avoidance algorithm. So I want to know how to output the depth to SITL to achieve this target. I cannot find the code part where the airsim output the depth/image info to the port.

I am very confusing. Could you please give some instructions? Thank you.

**Comment by rajat2004** [comment, marked as answer]
You can use the [Image APIs](https://microsoft.github.io/AirSim/image_apis/) to fetch the depth images, and then generate obstacle information in a format which ArduPilot, PX4 can receive, such as [OBSTACLE_DISTANCE](https://mavlink.io/en/messages/common.html#OBSTACLE_DISTANCE), or whichever other messages or capabilities are supported
